aten/src/ATen/FunctionalInverses.cpp

```diff
@@ -312,6 +313,77 @@ Tensor FunctionalInverses::_nested_view_from_jagged_inverse(const Tensor& base,
   }
 }
 
+Tensor FunctionalInverses::_nested_strided_to_jagged_inverse(const at::Tensor & base, const at::Tensor & mutated_view, at::functionalization::InverseReturnMode inverse_return_mode, const at::Tensor & dummy) {
+  // Mutated view is a jagged NT
+  auto values = at::_nested_get_values(mutated_view);
+  auto offsets = at::_nested_get_offsets(mutated_view);
+  auto lengths = at::_nested_get_lengths(mutated_view);
+  auto ragged_idx = at::_nested_get_ragged_idx(mutated_view);
+
+  // Create a new C++ NT from the Python NestedTensor
+  // Start by creating metadata needed by C++ NT
+  auto options = TensorOptions().dtype(kLong);
+  auto nested_sizes = at::empty({offsets.size(0) - 1, values.dim()}, options);
+
+  auto non_ragged_dims = at::empty({values.dim()});
+  auto non_ragged_sizes = at::empty({values.dim()});
+  auto curr_dim = 0;
+  for (int64_t i = 0; i < values.dim() + 1; ++i) {
+    if (i != ragged_idx) {
+      non_ragged_dims[curr_dim] = i;
```

Comment:
this loop should be pretty small but using the data_ptr can be surprsingly faster than the slicing

Reply:
agreed, in general it's way faster to deal with data_ptrs directly when operating on strided C++ NT metadata